drivers/flash/flash_ifx_cat1.c

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright 2023 Cypress Semiconductor Corporation (an Infineon company) or
+ * an affiliate of Cypress Semiconductor Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT	  infineon_cat1_flash_controller
+#define SOC_NV_FLASH_NODE DT_PARENT(DT_INST(0, fixed_partitions))
+
+#define PAGE_LEN   DT_PROP(SOC_NV_FLASH_NODE, erase_block_size)
+#define PAGE_LEN_P 9
```

Comment:
This is not ok. `PAGE_LEN` is picked from DTSI, but `PAGE_LEN_P` is hardcoded and independent from `PAGE_LEN`.
I have seen that the `PAGE_LEN_P` is used for masking offsets; if you want to have a mask rather do something like
`PAGE_MASK ((uint32_t)-1) & ~(PAGE_LEN - 1))`.